packages/outstatic/src/pages/add-custom-field/index.tsx

```diff
@@ -0,0 +1,557 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import camelCase from 'camelcase'
+import { useContext, useEffect, useState } from 'react'
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form'
+import capitalize from 'capitalize'
+import * as yup from 'yup'
+import { AdminLayout, Input } from '../../components'
+import Alert from '../../components/Alert'
+import Modal from '../../components/Modal'
+import { OutstaticContext } from '../../context'
+import { useCreateCommitMutation } from '../../graphql/generated'
+import { CustomField, CustomFields, customFieldTypes } from '../../types'
+import useFileQuery from '../../utils/useFileQuery'
+import useNavigationLock from '../../utils/useNavigationLock'
+import useOid from '../../utils/useOid'
+import { createCommit as createCommitApi } from '../../utils/createCommit'
+
+type AddCustomFieldProps = {
+  collection: string
+}
+
+type CustomFieldForm = CustomField & { name: string }
+
+export default function AddCustomField({ collection }: AddCustomFieldProps) {
+  const {
+    contentPath,
+    monorepoPath,
+    session,
+    repoSlug,
+    repoBranch,
+    repoOwner
+  } = useContext(OutstaticContext)
+  const [createCommit] = useCreateCommitMutation()
+  const fetchOid = useOid()
+  const [hasChanges, setHasChanges] = useState(false)
+  const [customFields, setCustomFields] = useState<CustomFields>({})
+  const yupSchema = yup.object().shape({
+    title: yup
+      .string()
+      .matches(/^[aA-zZ\s]+$/, 'Only alphabets are allowed for this field.')
+      .required('Custom field name is required.'),
+    type: yup.string().oneOf(['text', 'string']).required(),
+    description: yup.string(),
+    required: yup.boolean().required()
+  })
+  const [adding, setAdding] = useState(false)
+  const [error, setError] = useState('')
+  const methods = useForm<CustomFieldForm>({
+    mode: 'onChange',
+    resolver: yupResolver(yupSchema)
+  })
+  const [showAddModal, setShowAddModal] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
+  const [deleting, setDeleting] = useState(false)
+  const { data: schemaQueryData, loading } = useFileQuery({
```

Comment:
One observation. Custom fields will be unique to a collection, but Metadata queries will operate across all collections (and should be narrowed by collection if you care about a specific collection's custom fields). I don't think it will be an issue, but may be worth specifically noting in the documentation.